Fix compiler warning and init() behaviour

winkj opened the pull request:
Two bugfixes reported in ticket #16:
- unused variable marked as such to avoid compiler warning
- init() was updated to behave the same way whether the user called with in auto-detect mode, or with a fixed sensor type: previously, only the auto-detect case would also attempt to read from the sensor, now both code paths will do so

winkj committed:
fix compiler warning in SHTSensor::setAccuracy()

SHTSensor::setAccuracy(SHTAccuracy newAccuracy) is simply returning
false in the base class, since most of the sensors don't support
it. as such, 'newAccuracy' is newer used.

Mark it as unused to get rid of the compiler warning in restrictive
compilation setups

winkj committed:
check working sensor communication in init()

Previously, init() behaved differently depending on
whether SHTSensor was instantiated with an fixed sensor
type or not (auto-detect). In the auto-detect case, it would
trigger a 'readSample()' and thus ensure that the
sensor was communicating, while the fixed initialization
no such check was performed.

This change addressed this by calling readSample() in both cases,
and thus guaranteeing that if init() returns true, the sensor driver
was successfully created, and communication with the sensor on the bus
has been established

rnestler commented on SHTSensor.h:
```diff
@@ -157,7 +166,7 @@ class SHTSensorDriver
    * Returns false if the sensor does not support changing the accuracy
    */
-  virtual bool setAccuracy(SHTSensor::SHTAccuracy newAccuracy) {
+  virtual bool setAccuracy(__attribute__((unused)) SHTSensor::SHTAccuracy newAccuracy) {
```
I'm not sure how portable `__attribute__((unused))` is and if it works with all Arduino compilers.

winkj replied:
I figured Arduino is mainly g++, but valid concern. C++17 has http://eel.is/c++draft/dcl.attr.unused#2, but that may be too high of a bar

winkj replied:
should I just update this to ```SHTSensor::SHTAccuracy /* newAccuracy */```, which should fix the warning as well?

rnestler replied:
> I figured Arduino is mainly g++, but valid concern.

I'm not sure what compiler the ESP32 based Arduino's use, but AFAIK it's not gcc/g++.

> should I just update this to `SHTSensor::SHTAccuracy /* newAccuracy */`, which should fix the warning as well?

Would work, alternatively you can just do:
```cpp
   virtual bool setAccuracy(__attribute__((unused)) SHTSensor::SHTAccuracy newAccuracy) {
       (void)newAccuracy; // silence warning about unused argument
       return false;
   }
```

winkj committed:
Use non-GCC specific way to avoid unused warning

winkj committed:
remove redundant readSample() call

Since init() now calls readSample() also
for SHTSensor instances created with a fixed type,
we can remove the additional check in the
auto-detect loop
